hr_attendance_report_theoretical_time/models/resource_calendar_leaves.py

```diff
@@ -6,6 +6,9 @@ class ResourceCalendarLeaves(models.Model):
 
     @api.model
     def search(self, args, offset=0, limit=None, order=None, count=False):
+        """
+        This function is overrided in order to add the domain into searches
```

Comment:
A docstring should explain for what is used the code, no what code does, which is self-explanatory. Try with something like:
```
Inject extra domain when being called from _theoretical_hours method.
WARNING: This method should be called only once with that context
from a restricted piece of code, or can have side effects.
```